app/services/api/v1x0/catalog/approval_transition.rb

```diff
@@ -57,6 +60,12 @@ def mark_denied
           Rails.logger.info("Order #{@order_item.order_id} has been denied")
         end
 
+        def mark_errored
+          finalize_order
+          @order_item.update_message("error", "Order #{@order_item.order_id} has approval errors")
+          Rails.logger.error("Order #{@order_item.order_id} has failed")
+        end
+
         def finalize_order
           @order_item.update(:state => @state)
```

Comment:
Do we have the Order/OrderItem states enumerated somewhere?  Is it only in the openapi.json file?

Reply:
@gmcculloug Currently its only in the openapi.json, I think we should add an enum for it in app models

Reply:
Is the plan to handle that in a separate PR?

Reply:
@gmcculloug Yes I will create a separate PR for that